### PR TITLE
[common] refactor column_name for special cases

### LINKED ISSUE #800: Context Function broke the column name
**Describe the bug**

Context Function broke the column name. 

It'll confuse the clients (all kinds of drivers...)

**How to reproduce**

```
mysql> select version();
+----------------------------------------------------------------------------------------+
| version(FuseQuery v-0.1.0-5015d73(1.54.0-nightly-2021-06-09T12:08:09.543328298+00:00)) |
+----------------------------------------------------------------------------------------+
| FuseQuery v-0.1.0-5015d73(1.54.0-nightly-2021-06-09T12:08:09.543328298+00:00)          |
+----------------------------------------------------------------------------------------+
1 row in set (0.01 sec)

mysql> select database();
+-------------------+
| database(default) |
+-------------------+
| default           |
+-------------------+
1 row in set (0.01 sec)
```

We should refactor the column_name method in Expression for these special cases.

### PR DESCRIPTION
**Pull request by b41sh**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary

refactor `column_name` for special cases

## Changelog

- Bug Fix

## Related Issues

Fixes #800

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: common/planners/src/plan_expression.rs [hunk 1]
```diff
@@ -90,6 +90,11 @@ impl Expression {
     pub fn column_name(&self) -> String {
         match self {
             Expression::Alias(name, _expr) => name.clone(),
+            Expression::ScalarFunction { op, args: _ } => match op.as_ref() {
```
Comment (BohuTANG) [inline common/planners/src/plan_expression.rs:93]: It looks better to add a map to do it, key is function name and the value is the display name.

Comment (b41sh) [reply common/planners/src/plan_expression.rs:93]: ok

Comment (sundy-li) [inline common/planners/src/plan_expression.rs:93]: Better to use `Expression::ScalarFunction {op, ..}` ?

**Review by PsiACE — COMMENTED** [review]
good job, please also change the examples in docs

File: common/planners/src/plan_expression.rs (hunk 1 above)
Comment (b41sh) [reply common/planners/src/plan_expression.rs]: ok

**Comment by b41sh** [comment]
PTAL @BohuTANG @sundy-li @PsiACE

**Review by sundy-li — APPROVED** [review]
LGTM, thanks!

**Pull request merged**